src/config/config.c

```diff
@@ -0,0 +1,256 @@
+    return 0;
+}
+
+int assign_int(void* ptr, const struct struct_map *map, int map_size, const char* name, const char* value) {
+    for (int i = 0; i < map_size; i++) {
+        if (MATCH_KEY(map[i].str)) {
+            int size = map[i].offset >> 12;
+            int offset = map[i].offset & 0xFFF;
+            switch (size) {
+                case TYPE_S8:
+                    *((s8 *)((u8*)ptr + offset)) = atoi(value); break;
+                case TYPE_S16:
+                    *((s16 *)((u8*)ptr + offset)) = atoi(value); break;
+                case TYPE_S32:
+                    *((s32 *)((u8*)ptr + offset)) = atoi(value); break;
+
+                case TYPE_U8:
+                    *((u8 *)((u8*)ptr + offset)) = atoi(value); break;
+                case TYPE_U16:
+                    *((u16 *)((u8*)ptr + offset)) = atoi(value); break;
+                case TYPE_U32:
+                    *((u32 *)((u8*)ptr + offset)) = atoi(value); break;
+
+                case TYPE_COLOR:
+                    *((u16 *)((u8*)ptr + offset)) = get_color(value); break;
+                case TYPE_FONT:
+                    *((u8 *)((u8*)ptr + offset)) = FONT_GetFromString(value); break;
+                case TYPE_BUTTON:
+                    *((u8 *)((u8*)ptr + offset)) = get_button(value); break;
+                case TYPE_SOURCE:
+                    *((u8 *)((u8*)ptr + offset)) = get_source(value); break;
+
+                case TYPE_STR_LIST: {
+                    // get the list
+                    i++;  // next entry is additional info for string list
+                    const char* const *list = (const char* const *)map[i].str;
+                    unsigned length = map[i].offset;
+                    for (unsigned j = 0; j < length; j++) {
+                        if (list[j] && MATCH_VALUE(list[j])) {
+                            *((u8 *)((u8*)ptr + offset)) = j;
+                            return 1;
+                        }
+                    }
+                    printf("unknow value %s for %s\n", value, name);
+                    break;
+                }
+                case TYPE_STR_CALL: {
+                    i++;
+                    StringCallback *callback = (StringCallback*)map[i].str;
+                    for (unsigned j = 0; j < map[i].offset; j++)
+                    {
+                        if (mapstrcasecmp(name, callback(i)) == 0) {
+                            *((u8 *)((u8*)ptr + offset)) = j;
+                            return 1;
+                        }
+                    }
+                    printf("unknow value %s for %s\n", value, name);
+                    break;
+                }
+                default:
+                    printf("Unknown type: %d\n", size);
+                    break;
+            }
+            return 1;
+        }
+    }
+    return 0;
+}
+
+int write_int2(void* ptr, const struct struct_map *map, int map_size,
+    const u16* defaults, int defaults_size, FILE* fh) {
```

Comment:
can you rename this?  from its name i'd expect it to write a single int, but it actually writes a whole struct of them. 'write_ints_from_map' or something more descriptive

Reply:
Sure. I will do it later. assign_int -> parse_ini_map, write_int ->write_ini_map, write_int->write_ini_map_defaults